File: vscode4teaching-server/src/main/java/com/vscode4teaching/vscode4teachingserver/controllers/dtos/ExerciseDTO.java
package com.vscode4teaching.vscode4teachingserver.controllers.dtos;

import org.hibernate.validator.constraints.Length;

import javax.validation.constraints.NotEmpty;

public class ExerciseDTO {
    @NotEmpty(message = "Name cannot be empty")
    @Length(min = 3, max = 100, message = "Exercise name should contain between 3 and 100 characters")
    public String name;

    public boolean includesTeacherSolution;

    public boolean solutionIsPublic;

    public boolean allowEditionAfterSolutionDownloaded;
}

File: vscode4teaching-webapp/src/app/model/rest-api/exercise.dto.ts
import { CourseDTO } from "./course.dto";

export interface ExerciseDTO {
    id: number;
    name: string;
    course: CourseDTO;
    includesTeacherSolution: boolean;
    solutionIsPublic: boolean;
    allowEditionAfterSolutionDownloaded: boolean;
}
